### ISSUE TITLE
Writes to vault's storage folder still occur even in Read-Only mode

### ISSUE THREAD

**Issue by buddydvd** [cryptomator/cryptofs#53]
## Basic Info

I'm using macOS in version 10.14.4.
I'm running Cryptomator in version 1.4.9.

## Description

Here's an examples that shows how writes to vault's storage folder still occur when the vault is opened in *Read-Only* mode.

1. Create an empty vault.
2. Unlock the vault in *Read-Only* mode.
3. List the content of the vault's storage folder.
4. Attempt to save a file into (or access a non-existent file at) the root folder of this vault with the following file name:
`0123456789abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz.txt`
5. List the content of the vault's storage folder.
6. Compare the vault's storage folder at step 3 and at step 5.

## Expected

The content should match.

## Actual

The content doesn't match.

## Terminal Log

```
Davids-MacBook-Pro:test_vault_4 david$ tree
.
├── d
│   └── PV
│       └── HG6MYJAKHZ2GTPTRYGYVK35HNNZOYQ
├── m
├── masterkey.cryptomator
└── masterkey.cryptomator.bkup

4 directories, 2 files
Davids-MacBook-Pro:test_vault_4 david$ cat /Volumes/WRdujCzVRdKQ_0/0123456789abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz.txt
cat: /Volumes/WRdujCzVRdKQ_0/0123456789abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz.txt: No such file or directory
Davids-MacBook-Pro:test_vault_4 david$ tree
.
├── d
│   └── PV
│       └── HG6MYJAKHZ2GTPTRYGYVK35HNNZOYQ
├── m
│   ├── BE
│   │   └── KH
│   │       └── BEKHY2XVUJPRRNESGIZOABYQTOH5EQKG.lng
│   ├── H2
│   │   └── W5
│   │       └── H2W5QASJPSB4333DGYRSTDEM6IY6C5GE.lng
│   └── JS
│       └── 2T
│           └── JS2TLPOQTKJ5YEGV7OZ5WUSA7ADHVBWU.lng
├── masterkey.cryptomator
└── masterkey.cryptomator.bkup

10 directories, 5 files
```

**Comment by overheadhunter** [comment]
Related to #52

**Comment by overheadhunter** [comment]
This should now be fixed due to the changes in 086a4f4: Changes are only written to the `m` directory when a "write" happens (including creation of directories or symlinks).

However, due to the increased complexity we should add an additional check to this method to make sure we fail fast in case of bugs:

https://github.com/cryptomator/cryptofs/blob/086a4f44cc78c9f24be35bf3e64cbb9327515b05/src/main/java/org/cryptomator/cryptofs/LongFileNameProvider.java#L95-L110
